docs/docs/examples/guides/04-object-detection-example.md

```diff
@@ -0,0 +1,745 @@
+
+To configure the *YOLO Executor*, set the following parameters:
+
+- Turn on the **auto-configure** and **auto-activate** switches
+- Set `Model Path` to the `.onnx` file, e.g., `/files/yolo12n.onnx`
+- Set `Classes Path` to the yaml label file, e.g., `/files/coco.yaml`
+- Set `Rate` to 3 (works on most machines)
+
+The parameter `Confidence Threshold` is the minimum score a predicted bounding box must have to be considered a valid detection. `IOU Threshold` is used during Non-Maximum Suppression (NMS) to decide whether two bounding boxes represent the same object. For example, if `IOU threshold` is set to 0.5, any box that overlaps more than 50% with a higher-scoring box will be discarded. For now we leave them as default values.
+
+:::info
+
+**Converting YOLO Model files to ONNX format**
+
+The *YOLO Executor* component expects an ONNX file. These can be created from a pretrained [
+`.pt`](https://github.com/sunsmarterjie/yolov12) model (we use the lightweight version — YOLO12n).
+
+Run the following Python script (with [`ultralytics`](https://pypi.org/project/ultralytics/) installed) to convert a
+`.pt` model to `.onnx`:
+
+```python
+from ultralytics import YOLO
+model = YOLO("yolov12n.pt")
+model.export(format="onnx")  # creates 'yolov12n.onnx'
+```
+
+:::
+
+These YOLO models are pre-trained on coco - classes file for
+found [here](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/coco.yaml). A fine-tuned
+model may require a custom yaml file.
+
+### Running the Application
+
+Start the application in AICA Studio. Then open **RViz**: bottom-right gear icon → "Launch RViz", then RViz → Add → By
+topic → `/yolo_executor/annotated_image/Image` to view the YOLO model's annotated output. It should show the camera
+images with bounding boxes drawn around key objects in it. The bounding boxes are published on the
+`yolo_executor/bounding_boxes` topic as `std_msgs/msg/String` and can be viewed in AICA Studio in the "ROS Topics" tab.
+Changing the `Confidence Threshold` and `IOU Threshold` parameters will change how often and how many bounding boxes the
+model outputs.
+
+:::note
+
+Only users with a Linux host can visualize the robot with RViz. On macOS, AICA Launcher will not show the RViz option.
+
+:::
+
+## Tracking an Object with YOLO
+
+The bounding boxes generated by YOLO can be used to move a robot towards an object.
+
+### Creating a Custom Component to Estimate Position from Bounding Boxes
+
+We first need to create a custom component which can estimate a 3D pose from a position in the camera frame. More
+information about custom components can be found [here](https://docs.aica.tech/docs/category/custom-components/). We
+make 3 simplifying assumptions:
+
+1. Camera is fixed and angled downward
+2. Known camera position pointing down: `(x=0, y=0.6, z=0.6)`
+3. Objects are on a flat surface at `z=0`
+
+#### Set up the repository
```

Comment:
I am wondering here if we should actually make this repo in our github and offer a direct clone option. @domire8 @eeberhard